annubes/task.py

```diff
@@ -88,12 +87,12 @@ class Task(TaskSettingsMixin):
     session: dict[str, float] = field(default_factory=lambda: {"v": 0.5, "a": 0.5})
     stim_intensities: list[float] = field(default_factory=lambda: [0.8, 0.9, 1])
     stim_time: int = 1000
-    catch_prob: float = 0.5
+    catch_prob: float | int = 0.5
     shuffle_trials: bool = True
     max_sequential: int | None = None
 
     def __post_init__(self):
-        if not self.catch_prob >= 0 and self.catch_prob < 1:
+        if not isinstance(self.catch_prob, float | int) or not (0 <= self.catch_prob < 1):
```

Comment:
Last time I checked, type hints did not raise any errors, if this is what you meant. I added the check because the tests were passing if a string was given as input for `catch_prob`. But I agree that is weird that we check the type only for this one. Should I remove it?

Reply:
If I change this line to `if not 0 <= self.catch_prob < 1:` and run pytest, I get (I think the expected behavior):

> FAILED tests/test_task.py::test_init_catch_prob[None-None] - TypeError: '<=' not supported between instances of 'int' and 'NoneType'
> FAILED tests/test_task.py::test_init_catch_prob[0.6-0.60] - TypeError: '<=' not supported between instances of 'int' and 'str'
Is this different for you?

I would revert it and update the test to expect a TypeError rather than a ValueError.

Reply:
Yes I agree and I'll add a clearer message for the user